test/Npgsql.Tests/ReaderTests.cs

```diff
@@ -1643,18 +1711,33 @@ await using (var reader = await cmd.ExecuteReaderAsync())
                 var task = reader.ReadAsync(cancellationSource.Token);
                 cancellationSource.Cancel();
 
-                var processId = (await postmasterMock.WaitForCancellationRequest()).ProcessId;
-                Assert.That(processId, Is.EqualTo(conn.ProcessID));
+                Assert.AreEqual(IsMultiplexing, !conn.Connector!.UserCancellationRequested);
 
+                if (!IsMultiplexing)
+                {
+                    var processId = (await postmasterMock.WaitForCancellationRequest()).ProcessId;
+                    Assert.That(processId, Is.EqualTo(conn.ProcessID));
+                }
+
+                // With multiplexing we don't send a cancellation request.
+                // We stil send a response, so we will free the thread.
```

Comment:
It's a bit weird to send a QueryCanceled response although we haven't actually cancelled... I'm not sure it makes sense to actually run this test for multiplexing. We can simply have a non-mock test which cancels with multiplexing, and simply makes sure the query completes?